cloud-controller-manager/do/firewall_controller.go

```diff
@@ -101,37 +103,51 @@ func NewFirewallController(
 	fwManager firewallManager,
 	workerFirewallTags []string,
 	workerFirewallName string,
+	metrics metrics,
 ) *FirewallController {
 	fc := &FirewallController{
 		kubeClient:         kubeClient,
 		client:             client,
 		workerFirewallTags: workerFirewallTags,
 		workerFirewallName: workerFirewallName,
 		fwManager:          fwManager,
+		metrics:            metrics,
 	}
 
 	serviceInformer.Informer().AddEventHandlerWithResyncPeriod(
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: func(cur interface{}) {
 				ctx, cancel := context.WithTimeout(context.Background(), 90*time.Second)
 				defer cancel()
-				err := fc.ensureReconciledFirewall(ctx)
+				var err error
+				defer func(start time.Time) {
+					metrics.eventHandlerDuration.Observe(time.Since(start).Seconds())
```

Comment:
I think the usage of [`ObserveDuration`](https://godoc.org/github.com/prometheus/client_golang/prometheus#Timer.ObserveDuration) could simplify this further.